cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mysql/MySQLBackendImpl.java

```diff
@@ -329,135 +361,161 @@ public void expirateRecordsCache(long expirationTime) throws CygnusRuntimeError,
-        if (con != null) {
+        LOGGER.debug("Closing MySQL connection objects.");
+        if (stmt != null) {
             try {
-                con.close();
+                stmt.close();
             } catch (SQLException e) {
                 throw new CygnusRuntimeError("Objects closing error", "SQLException", e.getMessage());
             } // try catch
         } // if
 
-        if (stmt != null) {
+        if (con != null) {
             try {
-                stmt.close();
+                con.close();
             } catch (SQLException e) {
                 throw new CygnusRuntimeError("Objects closing error", "SQLException", e.getMessage());
             } // try catch
         } // if
+
     } // closeMySQLObjects
-    
+
     /**
-     * This code has been extracted from MySQLBackendImpl.getConnection() for testing purposes. By extracting it into a
-     * class then it can be mocked.
+     * This code has been extracted from MySQLBackendImpl.getConnection() for
+     * testing purposes. By extracting it into a class then it can be mocked.
      */
     protected class MySQLDriver {
-        
-        private final HashMap<String, Connection> connections;
+
+        private final HashMap<String, DataSource> connections;
         private final String mysqlHost;
         private final String mysqlPort;
         private final String mysqlUsername;
         private final String mysqlPassword;
-        
+        private final int maxPoolSize;
+
         /**
          * Constructor.
+         * 
          * @param mysqlHost
          * @param mysqlPort
          * @param mysqlUsername
          * @param mysqlPassword
          */
-        public MySQLDriver(String mysqlHost, String mysqlPort, String mysqlUsername, String mysqlPassword) {
+        public MySQLDriver(String mysqlHost, String mysqlPort, String mysqlUsername, String mysqlPassword, int maxPoolSize) {
             connections = new HashMap<>();
             this.mysqlHost = mysqlHost;
             this.mysqlPort = mysqlPort;
             this.mysqlUsername = mysqlUsername;
             this.mysqlPassword = mysqlPassword;
+            this.maxPoolSize = maxPoolSize;
         } // MySQLDriver
-        
+
         /**
          * Gets a connection to the MySQL server.
+         * 
          * @param dbName
          * @return
+         * @throws CygnusRuntimeError
          * @throws CygnusPersistenceError
          */
         public Connection getConnection(String dbName) throws CygnusRuntimeError, CygnusPersistenceError {
             try {
-                // FIXME: the number of cached connections should be limited to a certain number; with such a limit
-                //        number, if a new connection is needed, the oldest one is closed
-                Connection con = connections.get(dbName);
+                // FIXME: the number of cached connections should be limited to
```

Comment:
Este comentario de FIXME entiendo que ya se podria borrar, ¿no?

Reply:
The number of connections per database, is now limitted by the pool, but,
Are there any behaviour within we need to limit the number of databases (pools)?

ie, limit number when Cygnus tries to create a new database per entity type, or something like that.

If this is the case, It would be interesting to implement in this PR.